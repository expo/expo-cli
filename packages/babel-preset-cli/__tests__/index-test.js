const babel = require('@babel/core');
const { wrap } = require('jest-snapshot-serializer-raw');

const preset = require('..');

function transform(input, options = {}) {
  return wrap(
    babel.transform(input, {
      filename: 'file.ts',
      babelrc: false,
      presets: [preset],
      sourceMaps: true,
    }).code
  );
}

it('optional-chaining', () => {
  expect(
    transform(`
      foo?.bar;
    `)
  ).toMatchSnapshot();
});
it('nullish-coalescing', () => {
  expect(
    transform(`
      var foo = object.foo ?? "default";
    `)
  ).toMatchSnapshot();
});
it('class-properties', () => {
  expect(
    transform(`
      class Bork {
        //Property initializer syntax
        instanceProperty = "bork";
        boundFunction = () => {
          return this.instanceProperty;
        };
    
        //Static class properties
        static staticProperty = "babelIsCool";
        static staticFunction = function() {
          return Bork.staticProperty;
        };
      }
    `)
  ).toMatchSnapshot();
});
it('preset-typescript', () => {
  expect(
    transform(`
      const x: number = 0;
    `)
  ).toMatchSnapshot();
});
it('plugin-transform-modules-commonjs', () => {
  expect(
    transform(`
      export default 42;
    `)
  ).toMatchSnapshot();
});
