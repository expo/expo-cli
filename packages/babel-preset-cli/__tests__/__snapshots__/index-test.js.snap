// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class-properties 1`] = `
Object {
  Symbol(jest-snapshot-serializer-raw): "\\"use strict\\";

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

class Bork {
  constructor() {
    _defineProperty(this, \\"instanceProperty\\", \\"bork\\");

    _defineProperty(this, \\"boundFunction\\", () => {
      return this.instanceProperty;
    });
  }

}

_defineProperty(Bork, \\"staticProperty\\", \\"babelIsCool\\");

_defineProperty(Bork, \\"staticFunction\\", function () {
  return Bork.staticProperty;
});",
}
`;

exports[`nullish-coalescing 1`] = `
Object {
  Symbol(jest-snapshot-serializer-raw): "\\"use strict\\";

var _object$foo;

var foo = (_object$foo = object.foo) !== null && _object$foo !== void 0 ? _object$foo : \\"default\\";",
}
`;

exports[`optional-chaining 1`] = `
Object {
  Symbol(jest-snapshot-serializer-raw): "\\"use strict\\";

var _foo;

(_foo = foo) === null || _foo === void 0 ? void 0 : _foo.bar;",
}
`;

exports[`plugin-transform-modules-commonjs 1`] = `
Object {
  Symbol(jest-snapshot-serializer-raw): "\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.default = void 0;
var _default = 42;
exports.default = _default;",
}
`;

exports[`preset-typescript 1`] = `
Object {
  Symbol(jest-snapshot-serializer-raw): "\\"use strict\\";

const x = 0;",
}
`;
